### PR TITLE
Only show event reminders of events attendyes or attendmaybe

### PR DESCRIPTION
**Pull request by MrPetovan**
Part of #4513
Follow-up to #4515

See https://friendica.mrpetovan.com/display/c443a55c115ada60a05be0d632823611

I thought we had a method to get our own self-contact id (based on `local_user()` but I couldn't find it?

### CODE REVIEW

File: src/Model/Profile.php [hunk 1]
```diff
@@ -643,27 +643,29 @@ public static function getEventsReminderHTML()
 		$bd_format = L10n::t('g A l F d'); // 8 AM Friday January 18
 		$classtoday = '';
 
+		$self = dba::selectFirst('contact', ['id'], ['uid' => local_user(), 'self' => true]);
+
 		$s = dba::p(
 			"SELECT *
```
Comment (annando) [inline src/Model/Profile.php]: A ```SELECT *``` with a ```JOIN``` would add all the fields from the item table as well, I think. Some ```SELECT event.*``` would be better.

File: src/Model/Profile.php [hunk 2]
```diff
@@ -643,27 +643,29 @@ public static function getEventsReminderHTML()
...
 			FROM `event`
-			WHERE `event`.`uid` = ?
-			AND  `event`.`type` != 'birthday'
-			AND  `event`.`start` < ?
-			AND  `event`.`start` >= ?
-			AND NOT EXISTS (
-				SELECT `id`
-				FROM `item`
-				WHERE `item`.`uid` = `event`.`uid`
+			JOIN `item`
```
Comment (annando) [inline src/Model/Profile.php]: Could you use ```INNER JOIN```?

File: src/Model/Profile.php [hunk 3]
```diff
@@ -643,27 +643,29 @@ public static function getEventsReminderHTML()
...
 				AND `item`.`parent-uri` = `event`.`uri`
-				AND `item`.`verb` = ?
-				AND `item`.`visible`
-				AND NOT `item`.`deleted`
-			)
+			WHERE `event`.`uid` = ?
+			AND `event`.`type` != 'birthday'
+			AND `event`.`start` < ?
+			AND `event`.`start` >= ?
+			AND `item`.`author-id` = ?
```
Comment (annando) [inline src/Model/Profile.php]: The ```author-id``` (and ```owner-id```) are the contact id of the representing contact with the ```uid=0```.

Comment (MrPetovan) [reply src/Model/Profile.php]: I'm not sure what you're asking me to change here. The resulting attendance activity item has `uid=1` and `author-id=6`, which are respectively my user id and my self-contact id.

File: src/Model/Profile.php (hunk 2 above)
Comment (MrPetovan) [reply src/Model/Profile.php]: Gladly.

File: src/Model/Profile.php (hunk 1 above)
Comment (MrPetovan) [reply src/Model/Profile.php]: Well spotted.

File: src/Model/Profile.php (hunk 3 above)
Comment (annando) [reply src/Model/Profile.php]: If the "contact" record with "id=6" really has got the "uid=1" and "self=true" then your system is in trouble.

See here: https://github.com/friendica/friendica/blob/develop/src/Model/Item.php#L481

The "author-id" should always be a contact record with "uid=0".

Comment (MrPetovan) [reply src/Model/Profile.php]: Aaaah, I remember now the issue with matching a `local_user()` id with the associated public contact. I probably did wrong then.

Yep, it doesn't work.

Groumpf.

Comment (MrPetovan) [reply src/Model/Profile.php]: I KNEW SOMETHING EXISTED!

I wrote it: `public_contact()`.

**Pull request merged**